traversal/procedure/GraphProcedure.java

```diff
@@ -182,6 +178,21 @@ public Builder() {
             this.vertices = new HashMap<>();
         }
 
+        public static Builder create(GraphPlanner planner) {
+            Builder builder = new Builder();
+            Set<PlannerVertex<?>> registeredVertices = new HashSet<>();
+            Set<PlannerEdge.Directional<?, ?>> registeredEdges = new HashSet<>();
+            planner.vertices().forEach(vertex -> builder.registerVertex(vertex, registeredVertices, registeredEdges));
+            return builder;
+        }
+
+        public static Builder create(Structure structure, Map<Identifier, Integer> orders) {
```

Comment:
two ways of constructing a `GraphProcedure` exist now:
1. from a `GraphPlanner` - the normal way, but difficult to use in tests because we normally want an exact procedure!
2. from a `Structure`, where we provide an explicit ordering for each vertex - very useful for testing